crates/biome_js_analyze/src/semantic_analyzers/nursery/use_number_properties.rs

```diff
@@ -0,0 +1,195 @@
+        recommended: true,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+const GLOBAL_NUMBER_IDENTS: [&str; 4] = ["parseInt", "parseFloat", "NaN", "Infinity"];
+
+impl Rule for UseNumberProperties {
+    type Query = Semantic<AnyJsExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let model = ctx.model();
+        let (reference, global_ident_name) = global_identifier(node)?;
+        if !GLOBAL_NUMBER_IDENTS.contains(&global_ident_name.text()) {
+            return None;
+        }
+        model.binding(&reference).is_none().then_some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Use the "<Emphasis>"Number"</Emphasis>" properties instead of the global ones."
+                },
+            )
+            .note(markup! {
+                "ES2015 moved some globals into the "<Emphasis>"Number"</Emphasis>" properties for consistency."
+            }),
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let mut mutation = ctx.root().begin();
+        let (old_node, new_node) = match node {
+            AnyJsExpression::JsIdentifierExpression(expression) => {
+                let name = expression.name().ok()?.text();
+                if !GLOBAL_NUMBER_IDENTS.contains(&name.as_str()) {
+                    return None;
+                }
+                let replacement = match name.as_str() {
+                    "Infinity" => {
+                        let is_negative = expression
+                            .syntax()
+                            .parent()
+                            .and_then(|parent| {
+                                parent.children_with_tokens().find_map(|child| match child {
+                                    NodeOrToken::Token(token) => {
+                                        if token.kind() == JsSyntaxKind::MINUS {
+                                            Some(true)
+                                        } else {
+                                            None
+                                        }
+                                    }
+                                    NodeOrToken::Node(_) => None,
+                                })
+                            })
+                            .unwrap_or(false);
+                        if is_negative {
+                            "NEGATIVE_INFINITY"
+                        } else {
+                            "POSITIVE_INFINITY"
+                        }
+                    }
+                    _ => name.as_str(),
+                };
```

Comment:
Instead of checking every child and token, you could check if the parent is a unary operation. We could also handle the `+ Infinity` case.

```suggestion
                let (old_node, replacement) = match name.as_str() {
                    "Infinity" => {
                        if let Some(parent) = node.parent::<JsUnaryExpression>() {
                            match parent.operator.ok()? {
                                JsUnaryOperator::Minus => (AnyJsExpression::JsUnaryExpression(parent), "NEGATIVE_INFINITY"),
                                JsUnaryOperator::Plus => (AnyJsExpression::JsUnaryExpression(parent), "POSITIVE_INFINITY"),
                            }
                        } else {
                            (node.clone(), "POSITIVE_INFINITY")
                        }
                    }
                    _ => (node.clone(), name.as_str()),
                };
```